### PR TITLE
24.09 post

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

File: _posts/2024-09-01-openloco-v24.09.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+landscape generation and implemented various vehicle functions.
+
+For a complete summary of changes, please find the
+[OpenLoco changelog](https://github.com/OpenLoco/OpenLoco/releases/tag/v24.09) on GitHub.
+
+## Implement Update Track / Road Motion (#2174, #2588)
+
+Vehicles travel along tracks and road in much the same way. There are fixed lanes that they can
+travel with all positions along the lane precomputed. When a road vehicle needs to overtake it
+enters/leaves an overtake lane. Moving along a lane is controled by the UpdateMotion family of
```
Comment (LeftofZen) [inline _posts/2024-09-01-openloco-v24.09.md:21]: `controled` -> `controlled`

File: _posts/2024-09-01-openloco-v24.09.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+functions. With these functions a speed is converted into a distance travelled in a single update
+tick and then it iteratively looks up the new positions in the precomputed lane data (sub position
+data). In OpenRCT2 these functions were very complex and handled a lot of special case situations
```
Comment (LeftofZen) [inline _posts/2024-09-01-openloco-v24.09.md:24]: grammatical weirdness here. I'd just add a `.` after " ...special cast situations" and start the next line with a new sentence.

Comment (LeftofZen) [reply _posts/2024-09-01-openloco-v24.09.md:24]: or just add an `and` at the end of the line

File: _posts/2024-09-01-openloco-v24.09.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+the road paint functions represent a huge chunk ~3% of the vanilla game. Like the track paint
+functions they are repetitive and probably were automatically generated by some sort of code generator.
+Annoyingly they don't share enough to use the same code as we did for track paint but we could use
+the same technique (TestPaint) to generate our C++ version. With them implemented its another big
+jump in percentage completed and another step towards being able to add new road pieces.
+
+## Implement Create Airport Game Command (#2585, #2568, #2598, #2603)
+
+On the implementing game commands project this month we finished off airport creation and track
+removal. Airport creation a simple function that is very similar to the other building related
```
Comment (LeftofZen) [inline _posts/2024-09-01-openloco-v24.09.md:45]: "Airport creation [is] a simple function..."

File: _posts/2024-09-01-openloco-v24.09.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	 been using the game logic one for sounds. It doesn't really change anything but it would
+	 have caused issues when it came to multiplayer.
+  2. Large square stations couldn't be created. We had known for a few releases that our code for
+	 finding nearby stations wasn't quite correct and would prevent 16x16 square stations. It
+	 wasn't until Create Airport though until we had to solve that issue as it was causing a very
+	 minor desync.
+  3. Road vehicles may use the wrong track type when reversing. This is actually a vanilla mistake
+	 which we accidentally duplicated. Its unlikely to cause any real difference as it appears this
+     was only being looked at for working out max speeds on turns. But it is possible that this
+	 might cause a crash. Eitherway its now corrected. It might seem odd that we found this whilst
```
Comment (LeftofZen) [inline _posts/2024-09-01-openloco-v24.09.md:60]: "Either way"

File: _posts/2024-09-01-openloco-v24.09.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+@AaronVanGeffen implemented the track removal game command this month. Much more straight forward
+compared to track creation but still complex in its own ways. With this one implemented we now have
+only 14/84 left to implement. Four of which are road versions of track commands, four that look to
+be Ai only functions and the rest that are multiplayer related and might be able to be ignored.
+
+## Implement Map Generator Terrain In Random Areas (#2594)
+
+Another two functions from @AaronVanGeffen that finished off the last of the map generator! We
+now have complete control over map generation and @AaronVanGeffen has been using these new powers
+to start adding new features to the map generator (e.g. rivers, [ADD MORE]...).
```
Comment (LeftofZen) [inline _posts/2024-09-01-openloco-v24.09.md:76]: was the `ADD MORE` supposed to be filled in, or is it like a "we'll add more in this space in future"

**Review by LeftofZen — CHANGES_REQUESTED** [review]
couple minor spelling/grammar mistakes

**Comment by AaronVanGeffen** [comment]
To do:
- [ ] Address above review comments
- [ ] Mention new object editor release, https://github.com/OpenLoco/ObjectEditor/releases/tag/3.0.0
- [ ] Cover https://github.com/OpenLoco/OpenLoco/pull/2597
- [ ] Cover https://github.com/OpenLoco/OpenLoco/pull/2608

File: _posts/2024-09-01-openloco-v24.09.md (hunk 5 above)
Comment (duncanspumpkin) [reply _posts/2024-09-01-openloco-v24.09.md:76]: It was for @AaronVanGeffen to fill in with anything else

**Pull request merged**